### PR TITLE
feat: Get the best estimator from a comparison

### PR DESCRIPTION
**Pull request by auguste-probabl**
Implements `ComparisonReport.get_best_model`, which allows to extract the best report according to the given metric 

Example:
```python
from sklearn.datasets import make_classification
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from skore import ComparisonReport, EstimatorReport, train_test_split
X, y = make_classification(random_state=42)
split_data = train_test_split(X=X, y=y, random_state=42, as_dict=True)
report_1 = EstimatorReport(LogisticRegression(), **split_data)
report_2 = EstimatorReport(DecisionTreeClassifier(), **split_data)
report = ComparisonReport([report_1, report_2])
report.get_best_model(metric="accuracy")  # the default
```

### CODE REVIEW

**Comment by GaetandeCast** [comment]
Nice! As a next step we can try to find a way to determine if the advantage of the best model is statistically significant when we have the std deviation of the metric (comparison of cross vals). We can take inspiration from this scikit-learn example:
https://scikit-learn.org/stable/auto_examples/model_selection/plot_grid_search_stats.html#sphx-glr-auto-examples-model-selection-plot-grid-search-stats-py

**Comment by auguste-probabl** [comment]
For our consideration: https://scikit-learn.org/dev/auto_examples/model_selection/plot_grid_search_stats.html#comparing-two-models-frequentist-approach (implements the adjusted t-test mentioned [here](https://github.com/probabl-ai/skore/issues/1573#issuecomment-3628043580)). Thanks @ogrisel

**Comment by auguste-probabl** [comment]
This brings out a lot of questions.

Statistical estimators such as the corrected t-statistic mentioned above might not be appropriate for our use-case, especially considering that:
- The CrossValidationReports are fitted on the same data
- The CrossValidationReports might have been made with different cross-validation strategies (e.g. 3-fold and 5-fold, or stratified and non-stratified)

If it were possible to come up with a different statistic for every situation, this would still be suboptimal as it would expose users to a fair amount of complexity.

So maybe the best call is to use a simple strategy such as "Compute mean and standard deviation, and check if the error bars overlap", and be explicit that this method is not necessarily the best statistical test?

But then, what should we do if the bars do overlap? I don't see us raising an error saying "Couldn't get a statistically significant outcome", same for printing a warning along with the result. The UX would be poor for a result that is not worth it.

Further, getting the "best" estimator by 1 metric might not be a desirable thing for a data scientist; indeed, it's often more a question of tradeoffs, e.g. inference time vs. accuracy. It might be more interesting to show a "Pareto front" with the desired metrics and let the user make their choice of what is the best.

With that I'll close this and we'll go back to the drawing board for this feature.
